components/selectors/gecko_like_types.rs

```diff
@@ -6,10 +6,18 @@
 //! types in components/style/gecko/selector_parser.rs
 
 #[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Direction {
+    Rtl,
+    Ltr,
+    Other(String),
```

Comment:
This should still be `Box<[u16]>`, or `Box<str>` at least. That may avoid `Component` growing.